address/encoding.go

```diff
@@ -40,3 +40,26 @@ func compressedPubKeyDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error
 		val, "*btcec.PublicKey", l, btcec.PubKeyBytesLenCompressed,
 	)
 }
+
+func proofCourierAddrEncoder(w io.Writer, val any, buf *[8]byte) error {
+	if t, ok := val.(*ProofCourierAddr); ok {
+		addrBytes := []byte(*t)
+		return tlv.EVarBytes(w, &addrBytes, buf)
+	}
+	return tlv.NewTypeForEncodingErr(val, "*address.ProofCourierAddr")
```

Comment:
nit: can remove the `address.` part in the name to make it a bit shorter (that's just for eventual error messages).